locallib.php

```diff
@@ -518,6 +519,69 @@ function get_attempt($instanceid) {
     return $attempt;
 }
 
+/**
+ * Load all attempts from the database.
+ *
+ * @param int $instanceid - instance id of this digitala activity
+ * @return $attempts - object containing all attempt information
+ */
+function get_all_attempts($instanceid) {
+    global $DB;
+
+    $attempts = $DB->get_records('digitala_attempts', array('digitala'  => $instanceid));
+    return $attempts;
+}
+
+/**
+ * Load users name based on their id.
+ *
+ * @param int $id - id of the user
+ * @return $user - user object
+ */
+function get_user($id) {
+    global $DB;
+
+    $user = $DB->get_record('user', array('id' => $id));
+    return $user;
+}
+
+/**
+ * Load all attempts from the database.
+ *
+ * @param mixed $attempt - object containing attempt information
+ * @param int $instanceid - instance id of this digitala activity
+ * @param int $id - activity id
+ * @param int $d - course id
+ * @return $cells - cells containing table data
+ */
+function create_result_row($attempt, $instanceid, $id, $d) {
+    global $DB;
+
+    $digitalatype = $DB->get_record('digitala', array('id' => $instanceid));
+
+    $user = get_user($attempt->userid);
+
+    $username = new html_table_cell($user->username);
+    $type = new html_table_cell($digitalatype->attempttype);
+    if ($attempt->holistic) {
+        $score = new html_table_cell($attempt->holistic);
+    } else {
+        $score = new html_table_cell($attempt->gop_score);
+    }
+    $time = new html_table_cell('TODO');
+    $tries = new html_table_cell('TODO');
+
+    $urltext = page_url(3, $id, $d, $attempt->userid);
+    $urllink = html_writer::link($urltext, get_string('results_link', 'digitala'));
+    $reportlink = new html_table_cell($urllink);
+
+    $cell->attributes['class'] = 'showFine';
```

Comment:
Where is this variable `cell` defined? This is causing the bug shown in the image. It does function with commenting this row out.

Reply:
Oh yeah, this one is left there by mistake. And it turns out it was unnecessary too.